infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -439,13 +412,13 @@ public void clearLocalStreams() {
         // either on Source or Sink because we would be enforcing an update
         // without opening the stream, hence, leading to "apparent" data loss as
         // checkpoint won't run on these streams
-        Set<UUID> streamsToQuery = streamViewMap.keySet().stream()
+        Set<UUID> streamsToQuery = config.getStreamsIdToNameMap().keySet().stream()
                 .filter(id -> !replicatedStreamIds.contains(id) && !MERGE_ONLY_STREAMS.contains(id))
                 .collect(Collectors.toCollection(HashSet::new));
 
         log.debug("Total of {} streams were replicated from Source out of {}," +
             " sequencer query for {}, streamsToQuery={}",
-            replicatedStreamIds.size(), streamViewMap.size(),
+            replicatedStreamIds.size(), config.getStreamsIdToNameMap().size(),
```

Comment:
this log message will also be inaccurate now because config.getStreamsIdToNameMap() will only contain the "currently" opened streams.  You can reword it to something like this -
"Total of {} streams were replicated from Source, sequencer query for..."